cads_adaptors/adaptors/multi.py

```diff
@@ -24,9 +25,12 @@ def split_request(
             # get request values for that key
             req_vals = full_request.get(key, [])
             # filter for values relevant to this_adaptor:
-            these_vals = [
-                v for v in ensure_list(req_vals) if v in this_values.get(key, [])
```

Comment:
this line suggests that `req_vals` needs an `ensure_list` around it and lines like `if len(these_vals) > 0:` suggest `these_vals` is expected to be a list